### PR TITLE
feat: create github release plugin (SDKCF 4936)

### PR DESCRIPTION
**Pull request by leila-gu**
The plugin will create a new Github release and upload the modules mappings zip file to releases assets when the `./gradlew githubRelease` task is executed.

### CODE REVIEW

File: publish/releases/github-api.gradle [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    "Status $post.responseCode : $post.responseMessage\""
+            return response.id
+        } else {
+            throw new GradleException("Failed to create a release.\nServer returned status " +
+                    "$post.responseCode : $post.responseMessage")
+        }
+    }
+
+    private void uploadReleaseAssets(long releaseId) {
+        String baseUrl = "https://uploads.github.com"
```
Comment (vincentdelmundo) [inline publish/releases/github-api.gradle:89]: Is this the same URL for GHE server? Maybe we can just retrieve this from the response of `postRelease()` task? What do you think?

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just some minor comments.

File: publish/releases/github-api.gradle [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+         * If a file with the same name already exist,
+         * the task will fail with `Status: 422 Unprocessable Entity` server error.
+         */
+        uploadReleaseAssets(releaseId)
+
+    }
+
+
+    private long postRelease() {
+        String baseUrl = "https://api.github.com/repos"
```
[the remark this answers is not in the document]
Comment (leila-gu) [reply publish/releases/github-api.gradle]: it's working on GHE servers but Yes better to add it as env var.
Updated!

File: publish/releases/github-api.gradle (hunk 1 above)
Comment (leila-gu) [reply publish/releases/github-api.gradle]: Updated to use the response url.

**Review by vincentdelmundo — APPROVED** [review]
LGTM

**Pull request merged**